TopNews/app/src/main/java/com/fisfam/topnews/MainActivity.java

```diff
@@ -89,6 +96,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    //Choose country of news source
+    private void chooseCountry() {
+        mUserPref = new UserPreference(getApplicationContext());
+        // setup the alert builder
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Choose a country:");
+        // add a radio button list
+        String[] countries = {"Germany", "France", "England", "USA", "China", "Italy"};
+        //the button is always checked by the value stored in mUserPref
+        builder.setSingleChoiceItems(countries, Arrays.asList(countries).indexOf(mUserPref.getCountry()),
+                new DialogInterface.OnClickListener() {
```

Comment:
can be simplified to lambda

Reply:
T chua hoc toi lambda nen tat ca cac method nay t deu de nguyen de t de hieu hon